hoodie-client/src/test/java/com/uber/hoodie/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -1596,6 +1614,41 @@ public void testCommitWritesRelativePaths() throws Exception {
     }
   }
 
+  @Test
+  public void shouldCompactTableWithSpecificRecord() throws Exception {
+    HoodieWriteConfig config =
+        getConfigBuilder(SpecificRecordExampleValue.getClassSchema().toString(),
+            "example-specificvalue-table").build();
+    HoodieWriteClient client = new HoodieWriteClient<>(jsc, config);
+
+    insertNewDataSpecificRecord(client, 10);
+    // should compact small file
+    boolean success = insertNewDataSpecificRecord(client, 10);
+    assertTrue(success);
+  }
+
+  private boolean insertNewDataSpecificRecord(HoodieWriteClient client, int nb) throws IOException {
+    String commitTime = client.startCommit();
+    JavaRDD<WriteStatus> insert = client.insert(jsc.parallelize(generateInsertsSpecificRecord(nb)), commitTime);
+    return client.commit(commitTime, insert);
+  }
+
+  public List<HoodieRecord> generateInsertsSpecificRecord(int n) throws IOException {
```

Comment:
Any schema or data generators can be moved to `HoodieTestDataGenerator`

Reply:
done